libs/smart-ngrx/src/selector/virtual-array.class.ts

```diff
@@ -14,6 +14,7 @@ export class VirtualArray<
 > implements SmartArray<P, C>
 {
   rawArray: string[] = [];
+  fetchedIndexes: boolean[] = [];
```

Comment:
**suggestion (performance):** Consider using a more memory-efficient data structure for fetchedIndexes

While the boolean array is a good start, for very large arrays, you might want to consider using a more memory-efficient data structure like a BitSet or a sparse array implementation. This could significantly reduce memory usage for large VirtualArrays with sparsely fetched indexes.